### PR TITLE
[REVIEW PLS] Support Asset URL

### PR DESCRIPTION
**Pull request by HowlingEverett**

### CODE REVIEW

File: scripts/build/index.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ const {
   inputDir = join(process.cwd(), 'source'),
   outputDir = join(process.cwd(), 'dist'),
   basePath = '/',
+  assetPath = process.env.CDN_URL,
```
Comment (bradparker) [inline scripts/build/index.js]: Hrm ... I'm not stoked that the env var and the opt are different names?

Comment (bradparker) [inline scripts/build/index.js]: If we call it asset URL also ... then there's less ambiguity. It's gotta be a full URL please.

File: lib/prod-server.js [hunk 2]
```diff
@@ -13,13 +15,19 @@ module.exports = ({
...
+  basePath,
+  assetPath
 }) => {
   const baseMatch = new RegExp(`^${basePath}`)
   server.use(logger({
     format: '[%date - %time] %status %method %url'
   }))
-  server.use(serveStatic(staticDir))
+  if (assetPath) {
+    const mountPoint = urllib.parse(assetPath).path
```
Comment (bradparker) [inline lib/prod-server.js]: So this is what allows the server to act as an origin if needs be? Nifty. Still would prefer `assetUrl` ... even in this case (which, granted, is a little weird).

**Comment by bradparker** [comment]
I'll see i've got this.

My `basePath` is

```
/foo/
```

My `assetPath` is

```
https://foo.bar/baz/
```

My app is going to try and serve assets from

```
/baz/foo/
```

And they'll actually _be_ in `dist/foo/`

... I think I've missed something.

File: lib/prod-server.js [hunk 3]
```diff
@@ -1,5 +1,7 @@
 const { createServer } = require('http')
+const urllib = require('url')
```
Comment (bradparker) [inline lib/prod-server.js]: also 

```
const { parse: parseUrl } = require('url')
```

If you like.

Comment (HowlingEverett) [inline lib/prod-server.js]: Ah, not quite. I mean, I may have implemented it wrong, by my intention was for basePath and assetPath (Or `assetURL` then) to be separate.

So for the assetURL of `http://mycdnurl.com/baz`, and a basePath of `/foo/`, the connect server will serve static files from '/baz' and the basePath of `/foo/` will only apply to the application routing

Comment (HowlingEverett) [inline lib/prod-server.js]: So the serve scrip trims off the path from the absolute CDN url and uses it as the static origin. I'm not 100% sure if basePath is relevant, but I wanted to preserve its existing behaviour in PS static sites...

File: lib/prod-server.js (hunk 2 above)
Comment (HowlingEverett) [inline lib/prod-server.js]: I agree, assetURL is better, will change.

File: scripts/build/index.js (hunk 1 above)
Comment (HowlingEverett) [inline scripts/build/index.js]: I am also happy to make the env var ASSET_URL, I was just trying to meet Jon's request for 'I want to make it clear what we are setting this env var in config for'

**Pull request merged**